### PR TITLE
Feature - Marker arrows now match the hover styling of the edges.

### PR DESCRIPTION
**Pull request by JStuve**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
On hover of an edge, the marker arrow does not match the edge hover color. This was a feature request over in the JSON Crack repo: [JSON Crack Issue 236](https://github.com/AykutSarac/jsoncrack.com/issues/236)

reaflow-marker-before

Issue Number: N/A

## What is the new behavior?

On hover of an edge, the marker will now match the edge color for default, active and disabled edge states.

reaflow-marker-after

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Adding dynamic styling to marker ends does required individual markers to be defined. I was able to solve for the default behavior while allowing users to define their own hover marker for anything custom.

### CODE REVIEW

File: src/Canvas.tsx [hunk 1]
```diff
@@ -371,12 +383,30 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(
...
           onClick={onCanvasClick}
         >
-          {arrow !== null && (
+          {arrow !== null && arrowHover !== null && (
             <defs>
-              <CloneElement<MarkerArrowProps>
-                element={arrow}
-                {...(arrow as MarkerArrowProps)}
-              />
+              <React.Fragment key={`${id}-arrow`}>
```
Comment (amcdnl) [inline src/Canvas.tsx:388]: Let's make it just `Fragment` like:

```suggestion
              <Fragment key={`${id}-arrow`}>
```

File: src/symbols/Arrow/Arrow.module.css [hunk 2]
```diff
@@ -3,3 +3,15 @@
   shape-rendering: geometricPrecision;
   fill: #485a74;
 }
+
+.arrow__hover {
```
Comment (amcdnl) [inline src/symbols/Arrow/Arrow.module.css:7]: class names should be camel case

**Review by amcdnl — CHANGES_REQUESTED** [review]
I'm curious why replace the arrow completely on hover? Why not just have like `activeClassName` or something?

**Comment by JStuve** [comment]
> I'm curious why replace the arrow completely on hover? Why not just have like `activeClassName` or something?

That's a great question! It seems to be possible, however there is a lack of browser support for utilizing `fill: 'context-stroke'`, which would inherit the stroke of the path and ultimately make this solution much cleaner. [See Example 2](https://svgwg.org/svg2-draft/painting.html#VertexMarkerProperties).

The next best option is to create the individual marker definitions and referencing them by their unique id via the `marker-end` attribute. Since the marker ends are reference in the svg def, and not from an element within the path, targeting that element, from what I've seen, is only done through the attribute.

[Here is a good SO post on this](https://stackoverflow.com/questions/32964457/match-arrowhead-color-to-line-color-in-d3), one of the more recent comments circles back to the SVG documentation

**Pull request closed without merging**